lib/host_repository.py

```diff
@@ -97,6 +99,22 @@ def canonical_facts_host_query(account_number, canonical_facts):
     return find_non_culled_hosts(query)
 
 
+def system_profile_fact_host_query(account_number, system_profile_fact, value,
+                                   regex=None):
+    if regex:
+        logger.info("FINDING OSMAJOR (%s, %s)", regex, value)
+        logger.info("SYSTEMPROFILEFACTS 2 (%s)", Host.system_profile_facts[system_profile_fact].astext)
+        query = Host.query.filter(
+            (Host.account == account_number) &
+            (re.match(regex, Host.system_profile_facts[system_profile_fact].astext).group(2) == value)
+        )
+    else:
+        query = Host.query.filter(
+            (Host.account == account_number) & (Host.system_profile_facts[system_profile_fact].astext == value)
+        )
+    return find_non_culled_hosts(query)
```

Comment:
In this function the _Host.query.filter_ with the account condition repeats twice. The function essentially does two different things. Maybe we can have two functions instead. But one of them would be currently not used at all. So just keep the one?

Another option would be to just DRY the common part.

```suggestion
def system_profile_fact_host_query(account_number, system_profile_fact, value,
                                   regex):
    query = Host.query.filter(
        (Host.account == account_number) &
        (re.match(regex, Host.system_profile_facts[system_profile_fact].astext).group(2) == value)
        )
    return find_non_culled_hosts(query)
```